server/src/test/java/org/apache/druid/server/coordinator/duty/CompactSegmentsTest.java

```diff
@@ -2465,4 +2232,18 @@ public void testFindMaxNumTaskSlotsUsedByOneCompactionTaskWhenIsSequentialMode()
       Assert.assertEquals(1, CompactSegments.findMaxNumTaskSlotsUsedByOneCompactionTask(tuningConfig));
     }
   }
+
+  private static ArgumentCaptor<Object> setUpMockClient(final OverlordClient mockClient)
```

Comment:
Do we need to use a mock `OverlordClient`? We could try to reuse the `TestOverlordClient` already written here.

Reply:
I am not sure, I haven't studied this class enough to answer that question. The class was like this already: it used both `TestDruidLeaderClient` and a Mockito mock of `HttpIndexingServiceClient`. I was trying to update it to use `OverlordClient` without redesigning it. I'd like to leave it as-is, if you're OK with that.

Reply:
Yeah, makes sense. If needed, we can clean it up later.